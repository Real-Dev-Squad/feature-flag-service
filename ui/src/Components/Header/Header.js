import Image from '../Global/Image/Image';
import Button from '../Global/Button/Button';

export default function Header({ user }) {
  return (
    <header
      className='py-2 px-8 h-20 flex items-center bg-rds-primary text-slate-100'
    >
      <Image
        src={user.picture.url}
      />

      <h1 className='text-lg'>
        {user.github_display_name}
      </h1>

      <Button
        className={'ml-auto bg-white text-rds-primary hover:bg-slate-400'}
        text={'Logout'}
      />
    </header>
  )
}