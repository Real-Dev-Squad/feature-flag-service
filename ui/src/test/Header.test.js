import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom'
import Header from '../Components/Header/Header';
import { USER } from './Fixtures/user';

describe("Test Header component", () => {
    
    beforeEach(() => {
        render(<Header user={USER}/>);
    })

    it("Should have image tag", () => {
        const imageElement = screen.getByRole('img');
        expect(imageElement).toBeInTheDocument();
    })

    it("Should have src tag", () => {
        const MOCK_IMG_SRC = USER.picture.url;
        const imageElement = screen.getByRole('img', { 'src': MOCK_IMG_SRC });
        expect(imageElement).toBeInTheDocument();
    })

    it("Should have heading at top level", () => {
        const headingElement = screen.getByRole('heading', { level: 1 });
        expect(headingElement).toBeInTheDocument();
    })

    it("Should have heading username", () => {
        const username = USER.github_display_name;
        const headingElement = screen.getByRole('heading');
        expect(headingElement).toHaveTextContent(username);
    })

    it("Should have logout button", () => {
        const logoutButton = screen.getByRole('button');
        expect(logoutButton).toHaveTextContent('Logout');
    })

    it("Should match the snapshot", () =>{
      const { container } = render(<Header user={USER}/>);
      expect(container).toMatchSnapshot();
    })
})