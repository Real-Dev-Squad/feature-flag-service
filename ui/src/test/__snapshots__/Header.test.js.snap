// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Header component Should match the snapshot 1`] = `
<div>
  <header
    class="py-2 px-8 h-20 flex items-center bg-rds-primary text-slate-100"
  >
    <img
      class="rounded-full h-12 block mx-2"
      src="https://res.cloudinary.com/realdevsquad/image/upload/v1649837643/profile/EYNHBSK7riBKY775OOMB/mfzbgspqgvj6unz34unm.jpg"
    />
    <h1
      class="text-lg"
    >
      Pallab Sonowal
    </h1>
    <button
      class="py-2 px-4 rounded-lg ml-auto bg-white text-rds-primary hover:bg-slate-400"
    >
      Logout
    </button>
  </header>
</div>
`;

exports[`Test Header component should match the snapshot 1`] = `
<div>
  <header
    class="py-2 px-8 h-20 flex items-center bg-rds-primary text-slate-100"
  >
    <img
      alt="User Picture"
      class="rounded-full h-12 block mx-2"
      src="https://res.cloudinary.com/realdevsquad/image/upload/v1649837643/profile/EYNHBSK7riBKY775OOMB/mfzbgspqgvj6unz34unm.jpg"
    />
    <div
      aria-label="Username"
      class="text-lg"
      tabindex="0"
    >
      Pallab Sonowal
    </div>
    <button
      class="ml-auto bg-white py-2 px-4 rounded-lg text-rds-primary hover:bg-slate-400"
    >
      Logout
    </button>
  </header>
</div>
`;
